core/client/src/main/java/alluxio/client/block/RetryHandlingBlockWorkerClient.java

```diff
@@ -61,41 +65,50 @@
   private final InetSocketAddress mWorkerDataServerAddress;
   private final WorkerNetAddress mWorkerNetAddress;
   private final InetSocketAddress mRpcAddress;
-  private final ExecutorService mExecutorService;
-  private final HeartbeatExecutor mHeartbeatExecutor;
-  private final Future<?> mHeartbeat;
+  private static final ScheduledExecutorService HEARTBEAT_POOL = new ScheduledThreadPoolExecutor(
```

Comment:
(1) done.
(2) I didn't test the code with fewer threads. It takes time to run all the tests again. How about I push a PR to master. We can pick it up in 1.3.1. Or if there is other to run the tests again, I can include this one as well.
(3). Done.